src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -120,9 +164,9 @@ public void OnResponseHeader(ReadOnlySpan<byte> name, ReadOnlySpan<byte> value)
 
                 lock (SyncObject)
                 {
-                    if (_state != StreamState.ExpectingHeaders && _state != StreamState.ExpectingTrailingHeaders)
+                    if (_state == StreamState.ExpectingIgnoredHeaders)
```

Comment:
It seems like it would be better to do this check below, after we deal with pseudo-headers.